superset/assets/src/explore/components/controls/SubscriberLayer.jsx

```diff
@@ -0,0 +1,378 @@
+    });
+  }
+
+  handleColumnType(columnType, index) {
+    const subscriptionList = this.state.subscriptionList;
+    subscriptionList[index]['columnType'] = columnType;
+
+    let subscribe_columns = this.state.subscribe_columns;
+    subscribe_columns.length <= index ? subscribe_columns.push({}) : subscribe_columns;
+    subscribe_columns[index]['col'] = columnType;
+
+    this.setState({
+      columnType,
+      subscriptionList,
+      subscribe_columns,
+      allowMoreColumns: subscribe_columns[index]['col'] && subscribe_columns[index]['op'] ? true : false,
+    });
+  }
+
+  handleOperatorType(operatorType, index) {
+    let subscriptionList = this.state.subscriptionList;
+    subscriptionList[index]['operatorType'] = operatorType;
+
+    let subscribe_columns = this.state.subscribe_columns;
+    subscribe_columns.length <= index ? subscribe_columns.push({}) : subscribe_columns;
+    subscribe_columns[index]['op'] = operatorType;
+
+    this.setState({
+      operatorType,
+      subscriptionList,
+      subscribe_columns,
+      allowMoreColumns: subscribe_columns[index]['col'] && subscribe_columns[index]['op'] ? true : false,
+    });
+  }
+
+  addMoreColumns() {
+    const subscriptionList = this.state.subscriptionList;
+    const data = subscriptionList[subscriptionList.length - 1];
+    const newData = { columnType: '', operatorType: '', index: data.index + 1 }
+    this.setState(prevState => ({ subscriptionList: [...prevState.subscriptionList, newData] }))
+  }
+
+  removeColumn(e, column) {
+    this.setState(prevState => ({ subscriptionList: [...prevState.subscriptionList.filter(x => x.index !== column.index)] }))
+  }
+
+  deleteSubscriber() {
+    this.props.close();
+    if (!this.state.isNew) {
+      this.props.removeSubscriberLayer(this.state);
+    }
+  }
+
+  applySubscription() {
+    if (this.state.name.length && this.state.sliceId) {
+      const subscription = {};
+
+      Object.keys(this.state).forEach((k) => {
+        if (this.state[k] !== null) {
+          subscription[k] = this.state[k];
+        }
+      });
+
+      subscription['actions'] = [
+        "APPLY_FILTER"
+      ];
+
+      subscription['linked_slice'] = [
+        {
+          publisher_id: this.state.sliceId,
+          subscribe_columns: this.state.subscribe_columns,
+        }
+      ];
+
+      subscription['extras'] = this.state.extraValue;
+
+      this.props.addSubscriberLayer(subscription);
+      this.setState({ isNew: false, oldName: this.state.name });
+    }
+  }
+
+  submitSubscription() {
+    this.applySubscription();
+    this.props.close();
+  }
+
+  renderSingleSubscription(subscriptionData) {
+    const { columnType, operatorType, index } = subscriptionData;
+    const { allowColumnSelection, } = this.state;
+
+    const operators = this.getSupportedOperators();
+
+    return (
+      <div key={index} style={{ display: 'flex', flexDirection: 'row', justifyContent: 'space-between', width: '400px' }}>
+
```

Comment:
move this inline style to css

Reply:
there is not separate css for this popup so inline style